From 211af39aab2e322ad14fd7727977946b29fff3ba Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Fri, 6 May 2022 11:09:36 +0800
Subject: [PATCH] sw64: kapi: include generic modules.h
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 211af39aab2e322ad14fd7727977946b29fff3ba
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

This patch removes replicated macros which have been defined in
generic version.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/asm/module.h | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/arch/sw_64/include/asm/module.h b/arch/sw_64/include/asm/module.h
index 55e6e333585f..d1663aab4097 100644
--- a/arch/sw_64/include/asm/module.h
+++ b/arch/sw_64/include/asm/module.h
@@ -2,18 +2,12 @@
 #ifndef _ASM_SW64_MODULE_H
 #define _ASM_SW64_MODULE_H
 
+#include <asm-generic/module.h>
+
 struct mod_arch_specific {
 	unsigned int gotsecindex;
 };
 
-#define Elf_Sym		Elf64_Sym
-#define Elf_Shdr	Elf64_Shdr
-#define Elf_Ehdr	Elf64_Ehdr
-#define Elf_Phdr	Elf64_Phdr
-#define Elf_Dyn		Elf64_Dyn
-#define Elf_Rel		Elf64_Rel
-#define Elf_Rela	Elf64_Rela
-
 #define ARCH_SHF_SMALL	SHF_SW64_GPREL
 
 #ifdef MODULE
-- 
2.33.0

